jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/DisplayMessageFragment.java

```diff
@@ -1,17 +1,49 @@
 package com.e_conomic.jonfirstapp;
 
 import android.os.Bundle;
+import android.support.annotation.Nullable;
 import android.support.v4.app.Fragment;
 import android.view.LayoutInflater;
 import android.view.ViewGroup;
 import android.view.View;
+import android.widget.TextView;
+
 
 public class DisplayMessageFragment extends Fragment {
 
+    // The message to be displayed.
+    private String message = "";
+
+    // The TextView that displays the message.
+    private TextView display_message;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
```

Comment:
Almost, it is true that it indicates that the parameter could be null, but you won't be forced to handle it. Only thing to say about `super.onCreate(..)` is that that should ideally also declare the parameter as `@Nullable` There are tools like "FindBugs" that can use this annotation to give you static warnings if your do not check for null. Maybe the IDE is clever enough to also give you some help too. There is also `@NotNull`.

In Kotlin, however, you will be forced to check for null. Your code simply won't be able to compile if you don't. This it is not done with an annotation like here, but built into the type system as "Optionals". For example, here it would be an "Optional Bundle", and you will be forced to check if the "Optional Bundle" is a "Bundle" or "null".